lib/logthrdest/logthrdestdrv.h

```diff
@@ -284,7 +287,11 @@ log_threaded_dest_worker_flush(LogThreadedDestWorker *self, LogThreadedFlushMode
   glong request_latency = timespec_diff_msec(&now, &self->last_flush_time);
   self->last_flush_time = now;
 
-  stats_aggregator_add_data_point(self->owner->metrics.request_latency_hist, request_latency);
+  if (result == LTR_SUCCESS)
+    {
+      stats_aggregator_add_data_point(self->owner->metrics.batch_size_events_hist, batch_size);
+      stats_aggregator_add_data_point(self->owner->metrics.request_latency_hist, request_latency);
```

Comment:
Shouldn't we add a data point for request latency for non success cases, too? The server might reject the request, the latency is still important IMO.